### PR TITLE
Feature/pelagos 3400 tombstone

### PR DESCRIPTION
**Pull request by fightingtexasaggie**

### CODE REVIEW

File: src/Pelagos/Bundle/AppBundle/Controller/UI/TombstoneController.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            );
+            drupal_add_html_head($element, 'meta');
+            return $this->render(
+                'PelagosAppBundle:Tombstone:index.html.twig',
+                $twigData = array(
+                    'dataset' => $dataset,
+                )
+            );
+        } else {
+            throw $this->createNotFoundException("No pending state placeholder found for UDI: $udi");
```
[the remark this answers is not in the document]
Comment (wwnichols) [reply src/Pelagos/Bundle/AppBundle/Controller/UI/TombstoneController.php]: tombstone links should not exist/404 if dataset doesn't meet criteria for tombstone

File: src/Pelagos/Bundle/AppBundle/Resources/views/Tombstone/index.html.twig [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{% extends "PelagosAppBundle:template:UI.html.twig" %}
+
+{% block head %}
+    {{ parent() }}
+{% endblock %}
+
+{% block body %}
+    {% if dataset %}
+        <h2>
+            The dataset {{ dataset.udi }} is not available yet for download.
```
Comment (mickel1138) [inline src/Pelagos/Bundle/AppBundle/Resources/views/Tombstone/index.html.twig]: "not available yet" or "not yet available" ?

Comment (wwnichols) [reply src/Pelagos/Bundle/AppBundle/Resources/views/Tombstone/index.html.twig]: ticket says 'not available yet'
https://english.stackexchange.com/questions/147431/is-it-more-correct-to-say-you-have-not-yet-actioned-or-you-have-not-action

File: src/Pelagos/Bundle/AppBundle/Controller/UI/TombstoneController.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+     * @return Response
+     */
+    public function defaultAction($udi)
+    {
+        $dataset = $this->getDataset($udi);
+
+        if (($dataset->getAvailabilityStatus() == DatasetSubmission::AVAILABILITY_STATUS_PUBLICLY_AVAILABLE) or
+            ($dataset->getAvailabilityStatus() == DatasetSubmission::AVAILABILITY_STATUS_PUBLICLY_AVAILABLE_REMOTELY_HOSTED)) {
+            throw $this->createNotFoundException("No pending state placeholder found for UDI: $udi");
+        } else {
```
Comment (mickel1138) [inline src/Pelagos/Bundle/AppBundle/Controller/UI/TombstoneController.php]: This else is not really needed right?

Comment (fightingtexasaggie) [reply src/Pelagos/Bundle/AppBundle/Controller/UI/TombstoneController.php]: I guess not as the exception being thrown stops execution, but I left it in for clarity.

File: src/Pelagos/Bundle/AppBundle/Controller/UI/TombstoneController.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (fightingtexasaggie) [reply src/Pelagos/Bundle/AppBundle/Controller/UI/TombstoneController.php]: In production mode that message is not displayed. The response is always a 404.

File: src/Pelagos/Bundle/AppBundle/Controller/UI/TombstoneController.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+     *
+     * @Route("/{udi}")
+     *
+     * @return Response
+     */
+    public function defaultAction($udi)
+    {
+        $dataset = $this->getDataset($udi);
+
+        if ($dataset->getAvailabilityStatus() !== DatasetSubmission::AVAILABILITY_STATUS_PUBLICLY_AVAILABLE) {
```
[the remark this answers is not in the document]
Comment (fightingtexasaggie) [reply src/Pelagos/Bundle/AppBundle/Controller/UI/TombstoneController.php]: excluded I think you mean. Yes, public or remote public should both generate 404s.

File: src/Pelagos/Bundle/AppBundle/Controller/UI/TombstoneController.php (hunk 3 above)
Comment (fightingtexasaggie) [reply src/Pelagos/Bundle/AppBundle/Controller/UI/TombstoneController.php]: I've since simplified it.

**Pull request merged**